subsys/bluetooth/services/cts_client.c

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <sys/byteorder.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/gatt.h>
+
+#include <bluetooth/services/cts_client.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(cts_client, CONFIG_BT_CTS_CLIENT_LOG_LEVEL);
+
+/* The lowest valid Current Time year is the year when the Western calendar was introduced. */
+#define CTS_YEAR_MIN 1582
+/* The highest possible Current Time year. */
+#define CTS_YEAR_MAX 9999
+
+/* |     Year        |Month   |Day     |Hours   |Minutes |Seconds |Weekday |Fraction|Reason  |
+ * |     2 bytes     |1 byte  |1 byte  |1 byte  |1 byte  |1 byte  |1 byte  |1 byte  |1 byte  | = 10 bytes.
+ */
+#define CTS_C_CURRENT_TIME_EXPECTED_LENGTH 10
+
+int bt_cts_client_init(struct bt_cts_client *cts_c,
+		       bt_cts_client_evt_handler evt_handler)
+{
+	int err;
+
+	if (!cts_c || !evt_handler) {
+		return -EINVAL;
+	}
+
+	memset(cts_c, 0, sizeof(struct bt_cts_client));
+
+	cts_c->evt_handler = evt_handler;
+
+	err = k_sem_init(&cts_c->read_params_sem, 1, 1);
+
+	return err;
+}
+
+void bt_cts_client_on_disconnected(struct bt_cts_client *cts_c)
+{
+	k_sem_give(&cts_c->read_params_sem);
+}
+
+int bt_cts_client_handles_assign(struct bt_gatt_dm *dm,
+				 struct bt_cts_client *cts_c)
+{
+	const struct bt_gatt_dm_attr *gatt_service_attr =
+		bt_gatt_dm_service_get(dm);
+	const struct bt_gatt_service_val *gatt_service =
+		bt_gatt_dm_attr_service_val(gatt_service_attr);
+	const struct bt_gatt_dm_attr *gatt_chrc;
+	const struct bt_gatt_dm_attr *gatt_desc;
+
+	if (bt_uuid_cmp(gatt_service->uuid, BT_UUID_CTS)) {
+		return -ENOTSUP;
```

Comment:
EINVAL would be more appropriate.

Reply:
The other services return ENOTSUP at this point... You may check bas_client.c, dfu_smp.c, latency_client.c, nus_client.c, throughput.c.